protected.go

```diff
@@ -221,20 +206,15 @@ func (p *Protector) dialContext(op ops.Op, ctx context.Context, network, addr st
 	case <-ctx.Done():
 		return nil, ctx.Err()
 	default:
+		// continue
 	}
 
 	// Actually connect the underlying socket
-	err = conn.connectSocket()
+	err = conn.connectSocket(ctx)
```

Comment:
pass context down to `connectSocket` to remove the `select` below, and eliminate `connectTimeOut` altogether.